### DISCUSSION TITLE
[Bug] Dynamic Class

### DISCUSSION THREAD

**Discussion by be-bird-lade** [webstudio-is/webstudio-community#183] [Q&A]
If you assign a class to an element dynamically (in my case via a GraphQL API), it works in the editor, but after publishing the class is missing on the live page. It makes no difference whether you have atomic CSS enabled or not. If you set the class manually, it appears on the live page, it just doesn't seem to work with dynamic data.

**Comment by kof** [comment]
Sounds like you hit this bug https://github.com/webstudio-is/webstudio/issues/3683

As a workaround for now, can you use some data-attribute instead of class?

**Reply by be-bird-lade** [reply]
Great, thanks! I hadn't thought of that, but it works this way.
